swift/test/Ice/ami/AllTests.swift

```diff
@@ -276,56 +276,43 @@ func allTests(_ helper: TestHelper, collocated: Bool = false) async throws {
             await cb.value
         }
 
-        // TODO: Update to use async/await
-        //     do {
-        //         //
-        //         // Remote case.
-        //         //
-        //         let seq = ByteSeq(repeating: 0, count: 1024 * 10)
-
-        //         //
-        //         // Send multiple opWithPayload, followed by a close and followed by multiple opWithPaylod.
-        //         // The goal is to make sure that none of the opWithPayload fail even if the server closes
-        //         // the connection gracefully in between.
-        //         //
-        //         var maxQueue = 2
-        //         var done = false
-        //         while !done, maxQueue < 50 {
-        //             done = true
-        //             try p.ice_ping()
-        //             var results: [Promise<Void>] = []
-        //             for _ in 0..<maxQueue {
-        //                 results.append(p.opWithPayload(seq))
-        //             }
-
-        //             var cb = Promise<Bool> { seal in
-        //                 _ = p.close() {
-        //                     seal.fulfill($0)
-        //                 }
-        //             }
-
-        //             if try !cb.isResolved || cb.wait() {
-        //                 for _ in 0..<maxQueue {
-        //                     cb = Promise<Bool> { seal in
-        //                         results.append(p.opWithPayload(seq) { seal.fulfill($0) })
-        //                     }
-
-        //                     if try cb.isResolved && cb.wait() {
-        //                         done = false
-        //                         maxQueue *= 2
-        //                         break
-        //                     }
-        //                 }
-        //             } else {
-        //                 maxQueue *= 2
-        //                 done = false
-        //             }
-
-        //             for p in results {
-        //                 try p.wait()
-        //             }
-        //         }
-        //     }
+        do {
+            //
+            // Remote case.
+            //
+            let seq = ByteSeq(repeating: 0, count: 1024 * 10)
+
+            //
+            // Send multiple opWithPayload, followed by a close and followed by multiple opWithPayload.
+            // The goal is to make sure that none of the opWithPayload fail even if the server closes
+            // the connection gracefully in between.
+            //
+            var maxQueue = 2
+            while maxQueue < 50 {
+                try await p.ice_ping()
+
+                try await withThrowingTaskGroup(of: Void.self) { group in
+                    for _ in 0..<maxQueue {
+                        group.addTask {
+                            try await p.opWithPayload(seq)
+                        }
+                    }
+
+                    Task { try await p.closeConnection() }
```

Comment:
I would add this invocation to the task group. And if we expect an exception from the invocation, just catch and ignore it.